### PR TITLE
Fix ubuntu 22

### PR DESCRIPTION
**Pull request by grafin**
There where several errors, which prevented this action to be used in ubuntu:22.04 docker container. See commits for details.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -235,9 +235,13 @@ async function run_linux(): Promise<void> {
...
-      const files: Array<string> = output
-        .split('\n')
-        .filter(f => fs.statSync(f).isFile())
+      const files: Array<string> = output.split('\n').filter((f: string) => {
+        try {
+          return fs.statSync(f).isFile()
+        } catch (e) {
+          return false
+        }
+      })
```
Comment (Totktonada) [inline src/main.ts:244]: Am I understand right that here we catch an exception from `fs.statSync()` regarding the non-existent file? If so, it would be more accurate to pass the `throwIfNoEntry: false` [flag](https://nodejs.org/api/fs.html#fsstatsyncpath-options).

However, silent ignoring of all the non-existent files anyway looks a bit dangerous. I guess I can propose a better way.

See, the root of the problem is in `dpkg` excludes in the docker container.

```sh
$ cat /etc/dpkg/dpkg.cfg.d/excludes 
# Drop all man pages
path-exclude=/usr/share/man/*

# Drop all translations
path-exclude=/usr/share/locale/*/LC_MESSAGES/*.mo

# Drop all documentation ...
path-exclude=/usr/share/doc/*

# ... except copyright files ...
path-include=/usr/share/doc/*/copyright

# ... and Debian changelogs for native & non-native packages
path-include=/usr/share/doc/*/changelog.*
```

`apt-get install cmake` respects them, but `dpkg -L` doesn't. The file is created during build of the docker image, see [here](https://git.launchpad.net/livecd-rootfs/tree/live-build/auto/build?h=ubuntu/jammy#n179).

AFAIS, `dpkg -L` doesn't respect to explicitly passed `--exclude-path` and `--include-path` options. But we can just implement this filtering in Typescript.

I would hardcode the exclude/include rules as in the file listed above in the action code.

I don't insist, here are just my thoughts around.

However, I would describe the reason of the problem a bit deeper in the commit message (like it is done in this message, for example).

Comment (Totktonada) [reply src/main.ts:244]: I would also add that it is more related to run in a docker container job rather than Ubuntu of a particular version. I guess that it is as related to Ubuntu 22.04 as for 20.04.

File: src/main.ts [hunk 2]
```diff
@@ -197,7 +209,7 @@ async function run_linux(): Promise<void> {
 
     await core.group('Setting up repository', async () => {
       await exec.exec('sudo tee /etc/apt/sources.list.d/tarantool.list', [], {
-        input: Buffer.from(`deb ${baseUrl}/ubuntu/ ${distro} main\n`)
+        input: Buffer.from(`deb ${baseUrl}/${distro_id}/ ${distro} main\n`)
```
Comment (Totktonada) [inline src/main.ts:212]: I vote up this enhancement, but it doesn't look related to the pull request title: it is not about Ubuntu 22.04.

Since the pull request is more about docker container jobs rather than specific to Ubuntu 22.04 I would rephrase th pull request title/description.

**Comment by Totktonada** [comment]
@grafin, thanks for the patchset!

The changes are useful, look visually correct and AFAIK are tested on a private project.

However, there are some details I would polish before push.

First, pull request's title/description is a bit misleading: it seems, the changes are about docker container jobs in general rather than about specific Ubuntu 22.04. It would be nice to clarify.

Next, since our goal here is to make container jobs working, I would add some testing with `container: ubuntu:focal` into CI (and the same for `ubuntu:jammy` and for `debian:bullseye` and `debian:bookworm`). I guess that as result we'll add several more commits to fix remaining problems with container jobs and will get a ready to use action. At least, there is a known problem that we require `sudo` unconditionally even when the current user is the superuser (see #11 and #38).

**Comment by grafin** [comment]
Closed in favor of https://github.com/tarantool/setup-tarantool/pull/45 (new PR needed because I created a branch here to test new test:). I fixed all these review comments there.

**Comment by Totktonada** [comment]
@grafin Thanks! I'll take a look on a spare time.
